src/Controller/UserAccount/NotificationController.php

```diff
@@ -0,0 +1,51 @@
+<?php
+namespace App\Controller\UserAccount;
+
+use Doctrine\ORM\EntityManagerInterface;
+use App\Entity\Notifications\Notification;
+use App\Repository\NotificationRepository;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+class NotificationController extends AbstractController
+{
+    public function __construct(
+        private EntityManagerInterface $em,
+        private NotificationRepository $notificationRepository,
+    ) {
+    }
+
+    #[IsGranted('READ_NOTIFICATION', 'notification')]
+    #[Route(path: '/user/read-notification/{id}', name: 'read_notification', methods: ["POST"])]
+    public function readNotification(Notification $notification, Request $request): Response
+    {
+        if (!$this->isCsrfTokenValid('', $request->request->get('token'))) {
+            throw new AccessDeniedException();
+        }
+
+        $notification->setRead(true);
```

Comment:
```suggestion
        $notification->markAsRead();
```

and remove the setter on notification.